### ISSUE TITLE
[Question] Best practice to save and resume training with PPO + reward normalization

### ISSUE THREAD

**Issue by sky0470** [aai-institute/tianshou#846]
- [X] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] documentation request (i.e. "X is missing from the documentation.")
    + [ ] new feature request
- [X] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [X] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [x] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, gymnasium as gym, torch, numpy, sys
  print(tianshou.__version__, gym.__version__, torch.__version__, numpy.__version__, sys.version, sys.platform)
  ```
0.4.11 0.27.1 1.13.1+cu117 1.23.5 3.9.12 (main, Apr  5 2022, 06:56:58)
[GCC 7.5.0] linux

Currently I am doing some training with PPO. I use the following code to save checkpoints and resume training:

  ```
  def save_checkpoint_fn(epoch, env_step, gradient_step):
    # ...
    torch.save(
      {
        "model": policy.state_dict(),
        "optim": optim.state_dict(),
      }, ckpt_path
    )
    return ckpt_path
  ```

  ```
  policy = myPPOPolicy(...)
  checkpoint = torch.load(ckpt_path, map_location=args.device)
  policy.load_state_dict(checkpoint["model"])
  policy.optim.load_state_dict(checkpoint["optim"])
  ```

However, I have noticed that the training cannot resume properly. After some debugging, I think the problem is caused by reward normalization, since ```policy.state_dict()``` will not save the ```policy.ret_rms``` running mean/std of the policy.

In this case, should I save ```policy.ret_rms``` with pickle in ```save_checkpoint_fn```, and load it manually when resuming the run ? I have also thought of saving the whole policy instance directly with pickle, but I guess there might be some issue if I load a policy trained with gpu on a machine with no gpu ?

**Comment by Trinkle23897** [comment]
It's already in example, see https://github.com/thu-ml/tianshou/blob/4ac407c78f58102fa7f38ded6bfc1e42c703a4a7/examples/mujoco/mujoco_ppo.py#L168-L169 https://github.com/thu-ml/tianshou/blob/4ac407c78f58102fa7f38ded6bfc1e42c703a4a7/examples/mujoco/mujoco_ppo.py#L202-L204
